assignment-2/lean.md

```diff
@@ -1,3 +1,148 @@
 #Lean
 
-// TODO
\ No newline at end of file
+The LEAN methodology, also known as a *Toyota Production System*, is an
+engineering methodology which has its origin in the production method
+that Toyota used to produce its cars. Later, the book *The Goal* from
+Elia Goldratt extended the base of the methodology into a more
+generalist one. The methodology has spread to other areas, like medicine
+or software engineering, and many companies tried to replicate the
+process but so far none has been as successful as Toyota applying the
+methodology. \[1, pp 280\]
+
+The methodology aims to eliminate everything which is not necessary or
+does not have any value for the client, usually referred in general as
+“wastes”: overburden (*muri*), inconsistency (*mura*) and over waste
+(*muda*) in the process. Eliminating wastes is an effective way to
+increase efficiency. To achieve that, they use a system to produce the
+**exact** component, in the **exact** moment and the **exact** amount.
+
+The waste is classified in three different types:
+
+- *Muri* is the process overburden, doing things in a suboptimal way,
+  executing processes that take more resources or time than the
+  optimal one. It is solved by standardising the workflow optimally.
+- *Mura* is the inconsistency of the stock of products available. It
+  is solved by a JIT (Just In Time) inventory: there is only a little
+  or no inventory of products or components and when there is a
+  request of a product, the components are requested through a pull
+  system in which all the components are recursively requested until
+  they reach the external supplier.
+- *Muda* is the over waste in its more general meaning: all those
+  added processes or goods that are not request or have any value for
+  the customer. *Muda* can be divided into two types: *muda* type I,
+  which has no value for the customer but it is necessary for the
+  manufacturing of the product; and *muda* type 2, which has no value
+  for the customer and it is not necessary either. The aim is to
+  eliminate the *muda* type 2 but there is no straightforward way of
+  doing it because of the variety of things that can be
+  considered waste.
+
+The concept of waste is a traversal concept in other software
+development methodologies. *RUP* (Rational Unified Process) focuses on
+reducing risks and scrum focuses on delivering product. Both of them do
+not pay special attention to waste. Although scrum might look like aims
+to reduce waste as they only focus on delivering a product, their focus
+is to deliver whichever is the cost. This may lead to suboptimal
+practices.
+
+On one hand, RUP can be more wasteful than scrum. The early stages of
+RUP (business modelling, requirements and analysis & design) take a long
+period and effort and, although they produce deliverables and artefacts,
+they are not the requested one by the client. This qualifies as *muda*
```

Comment:
Only during the early stages. The product is not completed until more iterations are performed.